extension/observer/kafkatopicsobserver/extension.go

```diff
@@ -19,113 +18,109 @@ import (
 )
 
 var (
-	_ extension.Extension      = (*kafkaTopicsObserver)(nil)
-	_ observer.EndpointsLister = (*kafkaTopicsObserver)(nil)
```

Comment:
Any reason to remove the `EndpointLister`?

Reply:
Mostly because that's not the external API for observer extensions, `observer.Observable` is. The `EndpointsLister` is an interface between the implementation and `observer.EndpointsWatcher`, external clients don't need to have access to it.

Reply:
BTW I don't think the delineation between the external/internal API is particularly clear, I'll look into sending another PR to clarify the observer API.

Reply:
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/38416

Reply:
This change was moved to https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/38416, so resolving.